### ISSUE TITLE
[Bug] LR Scheduler misalignment with Gradient Accumulation

### ISSUE THREAD

**Issue by uygnef** [sgl-project/SpecForge#364]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/SpecForge/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

## Problem Description
The Learning Rate (LR) Scheduler currently steps based on `global_step` without accounting for `gradient_accumulation`. 

When `--draft-accumulation-steps` is enabled, the actual **model update frequency** (effective step) changes, but the scheduler logic remains tied to the raw step count. This leads to misaligned training curves—the model performance diverges when comparing runs with different accumulation settings (even if the effective batch size is theoretically the same).

## Context
In previous versions, the codebase distinguished between `draft_batch_size` and `micro_batch_size`. This explicit definition allowed users to control exactly how many samples constituted a parameter update. 

The current removal of this concept introduces ambiguity regarding the "step" definition in the LR scheduler.

## Proposed Solution
I suggest re-introducing a parameter similar to `draft_batch_size` (or `effective_batch_size`) to explicitly control the number of samples per model update.

**Benefits:**
1.  **Consistency:** Ensures the LR scheduler aligns with the actual model updates regardless of `grad_accum` settings.
2.  **Clarity:** Helps users debug and understand the training progress without ambiguity.

The relationship should hold:
$$\text{Effective Batch Size} = \text{Micro Batch Size} \times \text{Grad Accum Steps}$$

Using a fixed `draft_batch_size` as the anchor for the LR scheduler would resolve this misalignment.

### Reproduction

``` examples/run_llama3.1_8b_eagle3_offline.sh --output-dir outputs/qwen3-8b-eagle3-sharegpt_usp_grad_accum_enable_sdpa --draft-accumulation-steps 1 --batch-size 2 --sp-ulysses-size 1 --sp-ring-size 1 --num-epochs 10```

```examples/run_llama3.1_8b_eagle3_offline.sh --output-dir outputs/qwen3-8b-eagle3-sharegpt_usp_grad_accum_enable_sdpa --draft-accumulation-steps 2 --batch-size 1 --sp-ulysses-size 1 --sp-ring-size 1 --num-epochs 10```
have different lr

<img width="549" height="383" alt="Image" src="https://github.com/user-attachments/assets/9b57593c-f7c4-4619-be00-c376a298f025" />

### Environment

--
